paddle/cinn/backends/compiler.cc

```diff
@@ -29,13 +30,125 @@
 #endif
 
 DECLARE_string(cinn_source_code_save_path);
+DECLARE_string(cinn_dump_group_lowered_func);
+DECLARE_string(cinn_dump_group_source_code);
+DECLARE_string(cinn_dump_group_ptx);
+DECLARE_string(cinn_dump_group_instruction);
 
 namespace cinn {
 namespace backends {
 using ir::Module;
 
 static constexpr int DebugLogMaxLen = 30000;
 
+void DumpCompilationInfo::DumpLoweredFunc() {
+  if (FLAGS_cinn_dump_group_lowered_func.empty()) {
+    return;
+  }
+  for (int idx = 0; idx < info_.lowered_funcs.size(); idx++) {
+    auto dump_path = utils::StringFormat(
+        "%s/fusion_group_%d", FLAGS_cinn_dump_group_lowered_func.c_str(), idx);
+    if (!hlir::framework::MakeDirectory(
+            dump_path, S_IRWXU | S_IRGRP | S_IXGRP | S_IROTH | S_IXOTH)) {
+      LOG(WARNING) << "Failed to make directory: \"" << dump_path
+                   << "\", the lowered functions for this group will not dump.";
+    } else {
+      auto dump_file = utils::StringFormat(
+          "%s/%s", dump_path.c_str(), "lowered_function.txt");
+      VLOG(7) << "Dump lower functions to: " << dump_file;
+      std::ofstream of(dump_file, std::ios_base::out);
+      if (of.is_open()) {
+        of << info_.lowered_funcs[idx].front();
+      } else {
+        LOG(WARNING) << "Failed to open file: " << dump_file
+                     << ", please check your path.";
+      }
+      of.close();
+    }
+  }
+}
+
+void DumpCompilationInfo::DumpSourceCode() {
+  if (FLAGS_cinn_dump_group_source_code.empty()) {
+    return;
+  }
+  for (int idx = 0; idx < info_.source_codes.size(); idx++) {
+    auto dump_path = utils::StringFormat(
+        "%s/fusion_group_%d", FLAGS_cinn_dump_group_source_code.c_str(), idx);
+    if (!hlir::framework::MakeDirectory(
+            dump_path, S_IRWXU | S_IRGRP | S_IXGRP | S_IROTH | S_IXOTH)) {
+      LOG(WARNING) << "Failed to make directory: \"" << dump_path
+                   << "\", the source code for this group will not dump.";
+    } else {
+      auto dump_file =
+          utils::StringFormat("%s/%s", dump_path.c_str(), "source_code.cu");
+      VLOG(7) << "Dump source code to: " << dump_file;
+      std::ofstream of(dump_file, std::ios_base::out);
+      if (of.is_open()) {
+        of << info_.source_codes[idx];
+      } else {
+        LOG(WARNING) << "Failed to open file: " << dump_file
+                     << ", please check your path.";
+      }
+      of.close();
```

Comment:
Is this function has lots of common code with last function? If so, can we extract some common codes as util function?